analytics-core/src/main/java/com/segment/analytics/AnalyticsContext.java

```diff
@@ -115,23 +116,34 @@
    * {@link Context}. The {@link Analytics} client can be called from anywhere, so the returned
    * instances is thread safe.
    */
-  static synchronized AnalyticsContext create(Context context, Traits traits) {
+  static synchronized AnalyticsContext create(Context context, Traits traits,
+                                              boolean collectDeviceId) {
     AnalyticsContext analyticsContext =
         new AnalyticsContext(new NullableConcurrentHashMap<String, Object>());
     analyticsContext.putApp(context);
-    analyticsContext.putDevice(context);
+    analyticsContext.setTraits(traits);
+    analyticsContext.putDevice(context, collectDeviceId);
     analyticsContext.putLibrary();
     analyticsContext.put(LOCALE_KEY,
-        Locale.getDefault().getLanguage() + "-" + Locale.getDefault().getCountry());
+            Locale.getDefault().getLanguage() + "-" + Locale.getDefault().getCountry());
     analyticsContext.putNetwork(context);
     analyticsContext.putOs();
     analyticsContext.putScreen(context);
     putUndefinedIfNull(analyticsContext, USER_AGENT_KEY, System.getProperty("http.agent"));
     putUndefinedIfNull(analyticsContext, TIMEZONE_KEY, TimeZone.getDefault().getID());
-    analyticsContext.setTraits(traits);
     return analyticsContext;
   }
 
+  /**
+   * Create a new {@link AnalyticsContext} instance filled in with information from the given
+   * {@link Context}. The {@link Analytics} client can be called from anywhere, so the returned
+   * instances is thread safe. This is a convenience wrapper for the method
+   * {@link #create(Context, Traits, boolean)}
+   */
+  static synchronized AnalyticsContext create(Context context, Traits traits) {
```

Comment:
We don't need both the methods defined. Let's just use the one that keeps the `boolean` variant.